<a href="https://colab.research.google.com/github/EmanHrustemovic/FlyRank-AI-Intership-ML-Track-/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


**Lane 2: Refresh / Content Opportunity Scoring**

I selected Lane 2 because it directly maps to the
workflow demonstrated in the starter notebooks —
ranking pages by their need for review using
observable search and engagement signals. My
background in multi-task classification (Logistic
Regression, Random Forest, DistilBERT) from prior
NLP research makes this lane a natural fit. The
output — a ranked action queue with reason codes —
is immediately actionable and honest about what
the data can support.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision:** Which pages should a content reviewer
look at first for refresh or improvement?

**Who acts:** A content strategist or editor with
limited review capacity (~50 pages per week).

**Action:** Prioritize pages for refresh, expansion,
or monitoring based on evidence of decline or
missed opportunity.

**Cost of a wrong call:**
- False positive: reviewer wastes time on a page
  that does not need attention.
- False negative: a declining page is missed,
  losing traffic and visibility that could have
  been recovered.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

# Clone repo first if not already done
import os
if not os.path.exists('/content/flyrank'):
    os.system('git clone https://github.com/EmanHrustemovic/FlyRank-AI-Intership-ML-Track- /content/flyrank')

df = pd.read_csv('/content/flyrank/data/raw/content_refresh_anonymized.csv')

total = len(df)
declining = (df['trend_direction'] == 'down').sum()
pct_declining = (df['trend_direction'] == 'down').mean() * 100
avg_impressions = df['impressions_90d'].mean()
avg_ctr = df['ctr'].mean()

print(f"Total pages: {total}")
print(f"Declining pages: {declining}")
print(f"% declining: {pct_declining:.1f}%")
print(f"Avg impressions (90d): {avg_impressions:.0f}")
print(f"Avg CTR: {avg_ctr:.3f}")

Total pages: 30000
Declining pages: 16262
% declining: 54.2%
Avg impressions (90d): 5200
Avg CTR: 0.511


## 3. Quick look at the data (2-3 real numbers)

```python
Total pages: 30,000
Declining pages: 16,262
% declining: 54.2%
Avg impressions (90d): 5,200
Avg CTR: 0.511
```

**What the numbers say:**

Over half of all pages (54.2%) show a declining
trend signal — meaning more than 16,000 pages are
candidates for review. With an average of 5,200
impressions per page over 90 days, there is enough
visibility in the dataset to make prioritization
meaningful. An average CTR of 0.511 suggests room
for improvement across the inventory.

**Why this lane is worth 7 weeks:**

With 16,262 declining pages and limited reviewer
capacity, a ranked scoring model can help a content
team focus on the highest-priority pages first —
rather than reviewing at random or by gut feeling.
Even a modest improvement in Precision@50 (as
demonstrated in the starter pipeline: 0.24 → 0.74)
represents significant real-world value.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I CAN say:**
- "We observed that pages with X signal tend to
  appear more often in the declining group"
- "The model suggests these pages as priority
  review candidates"
- "This ranking is decision-support — a human
  reviewer makes the final call"

**What I CANNOT say:**
- "Refreshing this page will recover its traffic"
- "We proved what Google's algorithm prioritizes"
- "This model predicts future traffic with
  certainty"

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.